nx-dev/feature-package-schema-viewer/src/lib/schema-viewer.tsx

```diff
@@ -2,64 +2,121 @@ import {
   Lookup,
+  LookupResult,
 } from '@nrwl/nx-dev/data-access-packages';
 import { JsonSchema, JsonSchema1 } from '@nrwl/nx-dev/models-package';
 import { ParameterView } from './parameter-view';
 import { shouldShowInStage, Stage } from './stage';
 import { Type } from './types/type';
 
-export function SchemaViewer({
-  schema,
-  reference,
-  lookup,
-  stage,
-}: {
-  schema: JsonSchema1;
-  reference: string;
-  lookup: Lookup;
-  stage: Stage;
-}): JSX.Element {
-  const properties = schema.properties || {};
-  const isDeprecated = (schema: JsonSchema): boolean =>
-    typeof schema === 'boolean' ? false : !!schema['x-deprecated'];
-  const getAlias = (schema: JsonSchema): string =>
-    typeof schema === 'boolean' ? '' : (schema['alias'] as string);
+interface PropertySchema extends JsonSchema1 {
+  alias: string;
+  'x-deprecated': boolean;
+  'x-priority': 'important' | 'internal';
+}
+
+interface PropertyModel {
+  alias: string;
+  initialSchema: PropertySchema;
+  isRequired: boolean;
+  lookupResult: LookupResult;
+  propertyName: string;
+  propertyReference: string;
+}
 
-  const renderedProps = Object.keys(properties)
+const isPropertyDeprecated = (schema: PropertySchema | JsonSchema): boolean =>
+  typeof schema === 'boolean' ? false : !!schema['x-deprecated'];
+const getPropertyAlias = (schema: PropertySchema | JsonSchema): string =>
+  typeof schema === 'boolean' ? '' : (schema['alias'] as string);
+
+function getViewModel(
+  schema: JsonSchema1,
+  lookup: Lookup,
+  reference: string
+): PropertyModel[] {
+  const properties = schema.properties || {};
+  return Object.keys(properties)
     .sort((a, b) => a[0].localeCompare(b[0])) // Sort properties alphabetically
     .map((propertyName) => {
-      const propertySchema = properties[propertyName];
+      const propertySchema = properties[propertyName] as PropertySchema;
       const lookupResult = lookup.getSchema(propertySchema);
       return {
         propertyName,
-        alias: ((properties[propertyName] as any)['alias'] as string) ?? '',
+        alias: propertySchema['alias'] ?? '',
         initialSchema: propertySchema,
+        isRequired:
+          typeof schema.required !== 'undefined' &&
+          !!schema.required.find((n) => n === propertyName),
         lookupResult,
         propertyReference:
           lookupResult?.baseReference ||
           `${reference}/properties/${propertyName}`,
       };
-    })
-    .filter((p) => {
-      if (p.lookupResult === undefined) {
-        return true;
-      }
-      return shouldShowInStage(stage, p.lookupResult.schema);
-    })
-    .map((p) => {
-      const isRequired =
-        typeof schema.required !== 'undefined' &&
-        !!schema.required.find((n) => n === p.propertyName);
+    });
+}
 
+function extractPropertiesByImportance(properties: PropertyModel[]): {
+  required: PropertyModel[];
+  important: PropertyModel[];
+  internal: PropertyModel[];
+  rest: PropertyModel[];
+} {
+  return {
```

Comment:
I love filtering as much as anybody but this would probably be faster and simpler with a for loop, right?

Reply:
That's fair, will update!